### ISSUE TITLE
Hiera for Feature extraction 

### ISSUE THREAD

**Issue by mbilalshaikh** [facebookresearch/hiera#4]
Thank you for your amazing work. 

anyone has used hiera for feature extraction yet? If authors can provide some tips in this space would be highly appreciated.

**Comment by anshudaur** [comment]
@mbilalshaikh  you can check the inference notebook - 

# If you also want intermediate feature maps
_, intermediates = model(img_norm[None, ...], return_intermediates=True)

for x in intermediates:
    print(x.shape) -> the last one gives out features embeddings .

**Comment by dbolya** [comment]
Yup, that's correct. Just pass `return_intermediates=True` into the forward pass. It will return the feature map at the end of each stage in a list.

**Comment by gracikk-ds** [comment]
But what should I do in case of video inference? The outputs of the model look pretty weird.

Code example:
```python
# sub_batch: torch.Size([8, 3, 16, 224, 224])
# model: hiera_base_16x224 (mae_k400_ft_k400)
logits, embedding = self.model(sub_batch, return_intermediates=True)
    for emb in embedding:
        print(emb.shape)
```

Output:

```bash
torch.Size([8, 8, 56, 56, 96])
torch.Size([8, 8, 28, 28, 192])
torch.Size([8, 8, 14, 14, 384])
torch.Size([8, 8, 7, 7, 768])
```
@dbolya

**Comment by dbolya** [comment]
@gracikk-ds That is correct. The format is: `[batch, time, height, width, dim]`. There are half as many tokens across time because each token encodes 2 frames (each token is originally 2 frames x 4 pixel x 4 pixel). If you'd like to get information for all 16 frames, then interpolation is probably the way to go.

**Comment by dbolya** [comment]
@gracikk-ds Yes if you want 1 feature per clip that is correct. You may also want to normalize it with
```py
model.norm(emb)
```
depending on your application (only applies to the last 8x7x7 feature map).
